content/learning-paths/smartphones-and-mobile/profiling-unity-apps-on-android/6-unity-analyzer.md

```diff
@@ -0,0 +1,193 @@
+
+You can now see data for functions and markers with “collision” in their name.
+
+When you click on one of the functions or markers, you will see the time for that entry in the graph. The white bars in the graph are showing the total frame duration. The cyan color is the function or marker you have selected. This provides an immediate indication of what percentage this function or marker is taking.
+
+The Marker Summary panel is showing 24.06% for the marker named “Collision movement”.
+
+You can see the code from _CollisionCalculationScript.cs_ starting at line 256.
+
+```
+UnityEngine.Profiling.Profiler.BeginSample("Collision movement");
+collisionMoveMs.Start();
+CollisionMovement(numChar, characters);
+collisionMoveMs.Stop();
+UnityEngine.Profiling.Profiler.EndSample();
+```
+
+You can see we have used the BeginSample() and EndSample() functions to create a custom sample marker.
+
+## Compare data from pre- and post- optimization
+
+Up to now we have concentrated on a single data set - that of the unoptimized version. You will now import the optimized version and use the Analyzer to compare them.
+
+### Pull the optimized data into the Analyzer
+
+Keep the unoptimized data in the Analyzer. Leave the Name Filter set to “collision”.
+
+1. Click in the Profiler window or open it using the menu option Window/Analysis/Profiler
+
+1. Click Clear to reset the Profiler
+
+1. Click on the Load Binary (folder icon) in the top-right
+
+1. Select the data file you named neon.data
+
+1. Click back to the Analyzer window
+
+1. Now click Compare in the top-left. You will see the unoptimized data set repeated in two rows, each with a frame time graph. To the left of the graphs, you will see that both are labeled “Unsaved 1”
+
+1. Click on the Pull Data button for the second row.
+
+1. The graph will change to the new data set and will now be named “Unsaved 2”
+
+You are now looking at the unoptimized (upper graph) and optimized (lower graph).
+
+### Graph selection
+Each graph can have a range of frames selected. If you want to ensure that you select the same frames from both graphs, you can tick the option Pair Graph Selection. Do this now. When you click on a specific frame in either graph, the same frame will be selected in the other automatically.
+
+Right-click on either graph and choose Select All. Do the same for the lower graph.
+
+You are now comparing the first 300 frames of each data set.
+
+### Comparing the two data sets
+Immediately you will notice a difference in the “Top 10 markers on median frames” area. You can see the difference between the two median frames in the screenshot below. The upper frame duration is 87.7ms while the lower frame is 33.4ms (these numbers are to the right of the colored entries). The size of each marker/function for the optimized set are all reduced compared to the unoptimized version. As we would expect, since the optimizations will have reduced how long each of those operations takes.
```

Comment:
"This is as we would expect" to make complete sentence?

Reply:
done